### ISSUE TITLE
flagging stars

### ISSUE THREAD

**Issue by jlin0504** [jlin0504/GUESS#3]
I'm stuck trying to find a way to flag stars like 150706001601048 where the continuum curves downwards in CCD3. 

plotty plots:  https://www.dropbox.com/s/ntvkimnsrp2j3k9/ctm_param.pdf?dl=0

Help appreciated.

**Comment by ly-duong** [comment]
You could compare the continuum points with the normalised spectrum's median flux (or even just compare to 1) and if there are some minimum number of continuum points that is above/below the median by some percentage, flag the star.

**Comment by jlin0504** [comment]
I have tried this but it seems to work only for points above the continuum. Below the continuum there are spectral lines. I cant think of a way to distinguish them from bendy continuum.

**Comment by ly-duong** [comment]
Hi Jane,

My suggestion was to compare only continuum points to the median, not all data points. You shouldn't have this issue most of the time, although we do see that some continuum points fall on absorption lines in poorly reduced spectra. It's probably a good idea to flag these spectra, even if the normalisation works well anyway.

I don't know yet how to deal with cases where you have no continuum points in problematic areas (eg at the beginning of CCD4). We can say compare the first wavelength of the spectrum with the lowest continuum wavelength. If the difference is more than 5 - 10 A (I'm not really sure what this number should be) - we can choose the first few 2 - 3 pixels of the spectrum to compare to the median. Equally, we could also include these first 2 - 3 pixels as fake continuum points just to tie down the polynomial fit. I have suggested this fake continuum idea to Martin but he's not really keen, and I'm not sure if it'd even make a difference! It's up to you if you want to try it out.

 If I'm not clear just let me know

Cheers
Ly

-------- Message d'origine --------
De : Jane Lin notifications@github.com
Date : 28/02/2016 21:39 (GMT+10:00)
À : jlin0504/GUESS GUESS@noreply.github.com
Cc : Ly Duong ly.duong@anu.edu.au
Objet : Re: [GUESS] flagging stars (#3)

I have tried this but it seems to work only for points above the continuum. Below the continuum there are spectral lines. I cant think of a way to distinguish them from bendy continuum.

##

**Comment by martinasplund** [comment]
Hi Jane and Ly

If we want to cure the problem, I suspect that a higher order polynomial would
be necessary. If we are satisfied with just flagging them then Ly’s suggestion may
work. This problem only seems to arise occasionally for the red edge of CCD3 and 
the blue edge of CCD4 so a special treatment for those in terms of checks should
be sufficient rather than something general that checks everywhere for all CCDs.

Cheers

Martin

**Comment by martinasplund** [comment]
For CCD4 I think the easiest solution is to extend the wavelength region
slightly so that the continuum region at 7694A is always included for
reasonable RVs. Based on the measured RV of the star we can flag 
if any tellurics are predicted to fall exactly in that region of the spectrum
and/or have sigma clipping for that particular continuum window. We could 
also add another continuum region at 7697.3-7698.0 to furthermore
constrain the normalization in this region — this region is almost as good
as the other selected continuum windows. 

Cheers,

Martin

**Comment by jlin0504** [comment]
Hi Martin and Ly, do you think the wiggles in CCD3 and 4 will be fixed by the reduction team in the near future? If so then we should just flag them since we don't want to over fit the improved reduction.

**Comment by ly-duong** [comment]
We'd need to raise this issue with WG3, but I'm not sure if the situation will change very much. The wavy structure is to do with the way iraf is extracting the spectrum, I don't believe that it is a feature of the raw data. Since the reduction would take a long time to re-run, I think we should try to normalise/flag these spectra as best we can for the current reduction.

**Comment by jlin0504** [comment]
Hi, I spent sometimes testing out your ideas. The basic idea I had in mind was to count the number of flux points lying beyond a certain level of continuum. I have hand picked these stars as being abnormal to test the algorithm:
150831004001208,150706001601214,150706001601048,150703003101105,150427004801334,140707002101397,150719002901106,150719002901146,150719002901167,150718004401105,150718004401108,150606003401216,150606003401096,150603001801087,150603004801172,150405000901340,150405000901019,150405000901139,150405000901309,140805003601382,140805003601270,140809004201388,150603004301039,150603004801111,150603004801225,150606003401032,150606003401033,150606003401335,140805004801396,140809001601198,150719002901122,150830002301135,150901000601143

For using continuum regions instead of the whole spectrum: this misses stars which have well defined continuum regions but still with funny continua (e.g., 150703003101105, ccd1)

For using SNR dependent cutting: this is trickier than I thought. Initially I tried selecting stars which have n points where the flux > 1+x/SNR. I played around with different n and x but good-looking, high SNR stars always seem to be flagged falsely (e.g., 150703002601399). I found the hard cuts are still giving out the most similar results compared to the hand picked sample. 

Eventually I settled on this solution:
if CCD !=4 and (number of points >1.05)>150 points => flag.
if CCD ==4 and (number of points >1.05)>200 points => flag. (this accounts for the extra emission lines sometimes encountered in CCD4)

This method misses out on stars like 140707002101397, 150427004801334, where the continua are somewhat wavy but not enough to be >1.05. I hope Cannon will still be able to derive reasonable stellar parameters for stars like these. There is also one false detection 150827003401172, where I think CCD1 is kinda noisy. 

This is very much done by trial and error and is not perfect. But I think it serves the purpose of flagging potential funny looking stars reasonably well enough. I will push these changes to master shortly.
